UndertaleModTool/MainWindow.xaml.cs

```diff
@@ -101,6 +102,9 @@ public object Selected
 
         public List<Tab> ClosedTabsHistory { get; } = new();
 
+        private List<(GMImage, WeakReference<BitmapSource>)> _bitmapSourceLookup { get; } = new();
```

Comment:
is there a reason why this needs to be a list instead of a dictionary?

Reply:
So this was originally a dictionary, but I changed it to a list because every lookup does a full sweep on the weak references anyway. Not sure if there's any better way to do it, though (we could possibly defer the sweep operation).

Reply:
wdym with "every lookup does a full sweep on the references"?
The way I see it is that currently, it *always* iterates over the full list when you want to get just one item, cause it always invalidated everything.

With a dictionary, you would only check the image asked for, and only invalidate that.

Reply:
Yeah, so in this case it's weird, because we also want to clear the "dictionary" of the weak references that are no longer valid (because the GC freed their memory). If you don't do this, it could gradually leak memory, especially if the images are getting replaced by new ones. If we move this clearing process somewhere else, we don't need to do a linear lookup here, but then we have to find another good place to do that "sweep" operation anyway. As it stands, though, I don't think we ever have more than a small handful of textures loaded at once.

Reply:
I see, then probably no harm in keeping it as a list and keeping the linearity.